### PR TITLE
Supported type for all labels

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
image

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Review by klakhov — COMMENTED** [review]
Patch works for me. 
What do you think about adding appropriate `enum` for label types? It doesn't feel right to use consts like `'tag'` or `'any'`.

**Comment by bsekachev** [comment]
> Patch works for me. What do you think about adding appropriate `enum` for label types? It doesn't feel right to use consts like `'tag'` or `'any'`.

I think it makes sense..

File: cvat-ui/src/components/labels-editor/label-form.tsx [hunk 1]
```diff
@@ -78,7 +79,7 @@ export default class LabelForm extends React.Component<Props> {
             name: values.name,
             id: label ? label.id : idGenerator(),
             color: values.color,
-            type: label && label.id as number > 0 ? label?.type : values.type || label?.type || 'any',
+            type: values.type || label?.type || 'any',
```
Comment (klakhov) [inline cvat-ui/src/components/labels-editor/label-form.tsx:82]: `any` here should be replaced with `LabelType.ANY`

File: cvat-ui/src/components/labels-editor/label-form.tsx [hunk 2]
```diff
@@ -451,6 +452,37 @@ export default class LabelForm extends React.Component<Props> {
...
     }
 
+    private renderLabelTypeInput(): JSX.Element {
+        const { onSkeletonSubmit } = this.props;
+        const isSkeleton = !!onSkeletonSubmit;
+
+        const types = Object.values(LabelType)
+            .filter((type: string) => type !== LabelType.SKELETON);
+        const { label } = this.props;
+        const defaultType = isSkeleton ? 'skeleton' : 'any';
```
Comment (klakhov) [inline cvat-ui/src/components/labels-editor/label-form.tsx:462]: The same thing here

File: cvat-ui/src/components/annotation-page/standard-workspace/objects-side-bar/labels-list.tsx [hunk 3]
```diff
@@ -72,14 +75,20 @@ function LabelsListComponent(): JSX.Element {
             if (Number.isInteger(labelID) && label) {
                 if (Number.isInteger(activatedStateID)) {
                     const activatedState = states.filter((state: any) => state.clientID === activatedStateID)[0];
-                    if (activatedState) {
+                    const bothAreTags = activatedState.objectType === ObjectType.TAG && label.type === ObjectType.TAG;
+                    const labelIsApplicable = label.type === 'any' || activatedState.shapeType === label.type || bothAreTags;
```
Comment (klakhov) [inline cvat-ui/src/components/annotation-page/standard-workspace/objects-side-bar/labels-list.tsx:79]: Also here
```suggestion
                    const labelIsApplicable = label.type === LabelType.ANY || activatedState.shapeType === label.type || bothAreTags;
```

File: cvat-ui/src/components/annotation-page/standard-workspace/objects-side-bar/labels-list.tsx [hunk 4]
```diff
@@ -72,14 +75,20 @@ function LabelsListComponent(): JSX.Element {
...
+                    if (activatedState && labelIsApplicable) {
                         activatedState.label = label;
                         dispatch(updateAnnotationsAsync([activatedState]));
                     }
                 } else {
-                    dispatch(rememberObject({ activeLabelID: labelID }));
-                    message.destroy();
-                    message.success(`Default label was changed to "${label.name}"`);
+                    const bothAreTags = activeObjectType === ObjectType.TAG && label.type === ObjectType.TAG;
+                    const labelIsApplicable = label.type === 'any' || activeShapeType === label.type || bothAreTags;
```
Comment (klakhov) [inline cvat-ui/src/components/annotation-page/standard-workspace/objects-side-bar/labels-list.tsx:86]: Same here

**Review by klakhov — APPROVED** [review]

**Pull request merged**